### PR TITLE
Remove link definition from the menu tree storage on menu_link_content entity deletion

### PR DESCRIPTION
**Pull request by jeqq**

### CODE REVIEW

File: src/Entity/Storage/Sql/MenuLinkContentStorage.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+namespace Drupal\multiversion\Entity\Storage\Sql;
+
+use Drupal\multiversion\Entity\Storage\ContentEntityStorageInterface;
+
+/**
+ * Defines the controller class for menu_link_content.
+ */
+class MenuLinkContentStorage extends ContentEntityStorage implements ContentEntityStorageInterface {
```
Comment (dickolsson) [inline src/Entity/Storage/Sql/MenuLinkContentStorage.php]: Let's do the same thing here, as we did with the other entity storage classes. Extend the original core storage class, and apply the trait from Multiversion instead.

File: src/Entity/Storage/Sql/MenuLinkContentStorage.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+   */
+  public function delete(array $entities) {
+    parent::delete($entities);
+
+    /** @var \Drupal\Core\Menu\MenuLinkManagerInterface $menu_link_manager */
+    $menu_link_manager = \Drupal::service('plugin.manager.menu.link');
+
+    foreach ($entities as $menu_link) {
+      // Remove link definition from the menu tree storage.
+      $menu_link_manager->removeDefinition($menu_link->getPluginId(), FALSE);
```
Comment (dickolsson) [inline src/Entity/Storage/Sql/MenuLinkContentStorage.php]: Can we also add some sort of test for all of this, to ensure that this continues to work down the line?

File: src/Entity/Storage/Sql/MenuLinkContentStorage.php (hunk 1 above)
Comment (jeqq) [inline src/Entity/Storage/Sql/MenuLinkContentStorage.php]: @dickolsson We don't have a separate storage class for menu_link_content content type in core, it uses the SqlContentEntityStorage class.

Comment (dickolsson) [inline src/Entity/Storage/Sql/MenuLinkContentStorage.php]: Ok! :-) Let's just add tests for this change then.
On 11 Mar 2015 13:10, Andrei Jechiu <notifications@github.com> wrote:In src/Entity/Storage/Sql/MenuLinkContentStorage.php:
> @@ -0,0 +1,32 @@
> +<?php
> +
> +/**
> + * @file
> + * Definition of Drupal\multiversion\Entity\Storage\Sql\MenuLinkContentStorage.
> + */
> +
> +namespace Drupal\multiversion\Entity\Storage\Sql;
> +
> +use Drupal\multiversion\Entity\Storage\ContentEntityStorageInterface;
> +
> +/**
> + * Defines the controller class for menu_link_content.
> + */
> +class MenuLinkContentStorage extends ContentEntityStorage implements ContentEntityStorageInterface {

@dickolsson We don't have a separate storage class for menu_link_content content type in core, it uses the SqlContentEntityStorage class.

**Comment by dickolsson** [comment]
Sorry, I tried to answer your last comment via the Github email functionality. But it didn't format very nicely :P

**Pull request merged**